### PR TITLE
[nl] Add feedback link

### PR DESCRIPTION
**Pull request by chejennifer**
screen-recording-_74_

- also fixed an error about `<tr>` can not be a child of a `<td>`

### CODE REVIEW

File: static/js/apps/nl_interface/debug_info.tsx [hunk 1]
```diff
@@ -18,12 +18,20 @@
...
 import { Col, Row } from "reactstrap";
 
 import { DebugInfo, SVScores } from "../../types/app/nl_interface_types";
 
+const FEEDBACK_LINK =
+  "https://docs.google.com/forms/d/e/1FAIpQLSfqndIayVhN1bN5oeZT0Te-MhhBMBR1hn97Lgr77QTOpga8Iw/viewform?usp=pp_url";
+const QUERY_PARAM_PREFIX = "&entry.1322830239=";
+const SOURCE_PARAM_PREFIX = "&entry.1070482700=";
+const VERSION_PARAM_PREFIX = "&entry.1420739572=";
+const QUERY_CHAIN_PARAM_PREFIX = "&entry.1836374054=";
```
Comment (pradh) [inline static/js/apps/nl_interface/debug_info.tsx:33]: Document where these numbers come from?

**Review by pradh — COMMENTED** [review]
nice!

**Comment by shifucun** [comment]
Is the feedback for normal users? In that case, should it be under the chart box like it is in Google Search?

File: static/js/apps/nl_interface/debug_info.tsx (hunk 1 above)
Comment (chejennifer) [reply static/js/apps/nl_interface/debug_info.tsx:35]: done

**Comment by chejennifer** [comment]
> Is the feedback for normal users? In that case, should it be under the chart box like it is in Google Search?

I think first it will just be for the team, but I assume it will be for normal users as well. In the thread (on the core chat) about this, both options of in the debug panel and under the charts were suggested & debug panel seemed to be the one chosen.

File: static/js/apps/nl_interface/debug_info.tsx [hunk 2]
```diff
@@ -18,12 +18,22 @@
...
 
+import { debug } from "console";
 import _ from "lodash";
 import React, { useState } from "react";
 import { Col, Row } from "reactstrap";
 
 import { DebugInfo, SVScores } from "../../types/app/nl_interface_types";
 
+const FEEDBACK_LINK =
+  "https://docs.google.com/forms/d/e/1FAIpQLSfqndIayVhN1bN5oeZT0Te-MhhBMBR1hn97Lgr77QTOpga8Iw/viewform?usp=pp_url";
```
Comment (juliawu) [inline static/js/apps/nl_interface/debug_info.tsx:29]: Suggestion: can we also add a `JSON.stringify(DebugInfo)` pre-filled answer to the form?

File: static/js/apps/nl_interface/debug_info.tsx [hunk 3]
```diff
@@ -86,6 +98,26 @@ const matchScoresElement = (svScores: SVScores): JSX.Element => {
...
+    [QUERY_CHAIN_PARAM_PREFIX]: JSON.stringify(queryChain),
+  };
+  let link = FEEDBACK_LINK;
+  Object.keys(paramMap).forEach((prefix) => {
+    const value = paramMap[prefix];
+    if (!value) {
+      return;
+    }
+    // when prefilling google forms, need to replace space with "+"
+    link += `${prefix}${value.replaceAll(" ", "+")}`;
```
Comment (beets) [inline static/js/apps/nl_interface/debug_info.tsx:116]: consider calling encodeURI instead, which would also take care of other html special chars

**Review by beets — APPROVED** [review]
cool!

File: static/js/apps/nl_interface/debug_info.tsx (hunk 2 above)
Comment (chejennifer) [reply static/js/apps/nl_interface/debug_info.tsx:29]: I originally wanted to include debugInfo, but unfortunately it is too many characters to pass in the URL

Comment (pradh) [reply static/js/apps/nl_interface/debug_info.tsx:29]: debug-info might be too big.  its perhaps better suited for the BT log Bo has set up?

File: static/js/apps/nl_interface/debug_info.tsx (hunk 3 above)
Comment (chejennifer) [reply static/js/apps/nl_interface/debug_info.tsx]: done

**Pull request merged**